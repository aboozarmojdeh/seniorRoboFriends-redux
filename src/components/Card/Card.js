import React, { Fragment } from 'react';

const Card = ({ id, name, email, link }) => {
  return (
    <Fragment>
      <article className="dib mw5 bg-light-green br3 pa3 pa4-ns mv3 ba b--black-10 ma2 bw1 shadow-5 grow">
        <div className="tc">
          <img alt='card-img' src={`https://robohash.org/${id}?size=200x200`} className="br-100 h4 w4 dib ba b--black-05 pa2 bg-white" title="Photo of a kitty staring at you" />
          <h1 className="f3 mb2">{name}</h1>
          <h2 className="f5 fw4 mt0">{email}</h2>
        </div>
      </article>
    </Fragment>
  )
};

export default Card;
