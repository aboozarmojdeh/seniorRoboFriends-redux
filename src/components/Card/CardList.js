import React, { Fragment } from 'react';
import Card from './Card';
const CardList = ({ robots }) => {
    // if(true){
    //     throw new Error('NOOOOOOOOOOOO')
    // }
    const cardArray = robots.map((robot, i) => {
        return (
            <Card key={robot.id} id={robot.id} name={robot.name} email={robot.email} link={robot.link} />
        )
    })
    return (
        <Fragment>
        <div>
            {cardArray}
        </div>
        </Fragment>
    )
};

export default CardList;