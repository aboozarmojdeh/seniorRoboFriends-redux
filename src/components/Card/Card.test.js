import React from 'react';
import { shallow, mount, render } from 'enzyme';
// import renderer from 'react-test-renderer';
import Card from './Card';

// npm test -- --coverage --watchAll=false



test('expect to render Card component',()=>{
    const wrapper = shallow(<Card />); 
    expect(wrapper).toMatchSnapshot();
})

// it('expect to render Card Component', () => {
//     const tree = renderer.create(<Card />).toJSON();
//     console.log(tree)
//     expect(tree).toMatchSnapshot();
//   });
