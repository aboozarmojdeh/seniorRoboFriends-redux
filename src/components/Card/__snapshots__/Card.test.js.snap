// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<Fragment>
  <article
    className="dib mw5 bg-light-green br3 pa3 pa4-ns mv3 ba b--black-10 ma2 bw1 shadow-5 grow"
  >
    <div
      className="tc"
    >
      <img
        alt="card-img"
        className="br-100 h4 w4 dib ba b--black-05 pa2 bg-white"
        src="https://robohash.org/undefined?size=200x200"
        title="Photo of a kitty staring at you"
      />
      <h1
        className="f3 mb2"
      />
      <h2
        className="f5 fw4 mt0"
      />
    </div>
  </article>
</Fragment>
`;
